adafruit_requests.py

```diff
@@ -441,13 +451,70 @@ def _send_request(
         if data and isinstance(data, str):
             data = bytes(data, "utf-8")
 
+        if data is None:
+            data = b""
+
         self._send_as_bytes(socket, method)
         self._send(socket, b" /")
         self._send_as_bytes(socket, path)
         self._send(socket, b" HTTP/1.1\r\n")
 
         # create lower-case supplied header list
         supplied_headers = {header.lower() for header in headers}
+        boundary_str = None
+
+        # pylint: disable=too-many-nested-blocks
+        if files is not None and isinstance(files, dict):
+            boundary_str = _generate_boundary_str()
+            content_type_header = f"multipart/form-data; boundary={boundary_str}"
+
+            for fieldname in files.keys():
+                if not fieldname.endswith("-name"):
+                    if files[fieldname][0] is not None:
+                        file_content = files[fieldname][1].read()
```

Comment:
@FoamyGuy I have something similar in progress, but chunked out the read during the send, so it allows for sending large files. What are your thoughts?

Reply:
@justmobilize is that functionality that is supported by CPython requests too? Is it the same as this? https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding#chunked_encoding or am I misunderstanding that? 

Generally I'd say I'm in favor of any functionality that also exists in CPython unless it happens to require substantially large code size for marginal benefits.

I don't have any real experience with the chunked transfers, and the main project that I had in mind for using this functionality is uploading relatively small files so it wasn't something I attempted to implement or looked into but that doesn't mean I'm opposed to it.

Reply:
So it's more just internal to how we do it. So instead of building one huge data object that goes to:
```py
self._send(socket, bytes(data))
```
We read the files there in chunks and then send. So we're never using more than 100 (or whatever makes sense) bytes at a time.

Reply:
It sounds like a good improvement to me, but I don't have a clear idea how the code would look to make that work without interrupting the non-file upload requests and with requests that inlcude multiple "parts" where some of them are plain form values like `"othervalue": (None, "HelloWorld"),` from the example. 

If you've got an idea of how that would work or even have it started already I'd say feel free to add commit(s) to this branch to adapt it to work like that, or if you'd prefer you can fork this and link me to your fork with those changes and I can merge it back to here. Or it could always be a follow up PR. I don't have any strong preference, whatever is easiest for you is good with me.

Reply:
I'm working on some other PRs this afternoon. I'll fork this and move some things around and share it

Reply:
@FoamyGuy merged what you have and what I had and drafted a PR here: https://github.com/FoamyGuy/Adafruit_CircuitPython_Requests/pull/1

Added some comments of things it still needs

Let me know your thoughts